### PR TITLE
Add test affected_chosts_by_erratum_restrict_flag

### PR DESCRIPTION
**Pull request by swadeley**
This implements stubbed out test for restrict flags.

I changed the repo for the class to FAKE_9_YUM_REPO. This was to make use of some of the constants related to that repo. That means I need to test the whole class.

Thank you

### CODE REVIEW

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/cli/test_errata.py -k test_positive_list_affected_chosts_by_erratum_restrict_flag
================= test session starts ==============================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/virtwho/robottelo
plugins: cov-2.8.1, services-1.3.1, mock-1.10.4, forked-1.1.3, xdist-1.31.0
collecting ... 2020-05-20 21:39:13 - conftest - DEBUG - Collected 34 test cases
2020-05-20 22:39:14 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fc24e56d450
2020-05-20 22:39:14 - robottelo.ssh - INFO - Connected to [dell-r330-12.gsslab.brq.redhat.com]
2020-05-20 22:39:14 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-05-20 22:39:16 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-7.el7sat.noarch

2020-05-20 22:39:16 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fc24e56d450
2020-05-20 22:39:16 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-05-20 22:39:16 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 34 items / 33 deselected / 1 selected                                                                  

tests/foreman/cli/test_errata.py .                                                                         [100%]

=============== 1 passed, 33 deselected in 712.34 seconds ====================


robottelo]$ pytest tests/foreman/cli/test_errata.py -k HostCollectionErrataInstallTestCase
================= test session starts ===================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/virtwho/robottelo
plugins: cov-2.8.1, services-1.3.1, mock-1.10.4, forked-1.1.3, xdist-1.31.0
collecting ... 2020-05-20 21:39:21 - conftest - DEBUG - Collected 34 test cases
2020-05-20 22:39:22 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fca4d34b150
2020-05-20 22:39:22 - robottelo.ssh - INFO - Connected to [dell-r330-12.gsslab.brq.redhat.com]
2020-05-20 22:39:22 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-05-20 22:39:24 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-7.el7sat.noarch

2020-05-20 22:39:24 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fca4d34b150
2020-05-20 22:39:24 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2020-05-20 22:39:24 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 34 items / 22 deselected / 12 selected                                                                 

tests/foreman/cli/test_errata.py ............                                                              [100%]

================ 12 passed, 22 deselected in 5593.23 seconds ===================

**Comment by swadeley** [comment]
I will do a rebase after some reviews.

File: tests/foreman/cli/test_errata.py [hunk 1]
```diff
@@ -459,6 +466,149 @@ def test_negative_install_by_hc_name_without_org_info(self):
...
+
+
+        :expectedresults: List of affected content hosts for an erratum is
+            displayed filtered with corresponding restrict flags.
+
+        :CaseAutomation: automated
+
+        """
+        # Create list of uninstallabe errata by removing FAKE_2_ERRATA_ID
+        UNINSTAllABLE = FAKE_9_YUM_ERRATUM[1:5]
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_errata.py:503]: Can we make that more explicit and not depend of order of items in constant?
```python
UNINSTAllABLE = [erratum for erratum in FAKE_9_YUM_ERRATUM if erratum != FAKE_2_ERRATA_ID]
```
Or use set operations, but mind that will shuffle order of items inside. I think it doesn't matter in this case, but I'm adding that to let you know.
```python
UNINSTAllABLE = set(FAKE_9_YUM_ERRATUM) - set(FAKE_2_ERRATA_ID)
# just to let you know, you can wrap right side in `list()` if something down the line requires list and not any iterable; in this case, you don't need this
UNINSTAllABLE = list(set(FAKE_9_YUM_ERRATUM) - set(FAKE_2_ERRATA_ID))
```
Also, why double-L is lowercase in variable name?

File: tests/foreman/cli/test_errata.py [hunk 2]
```diff
@@ -459,6 +466,149 @@ def test_negative_install_by_hc_name_without_org_info(self):
...
+        erratum_list = Erratum.list(
+            {
+                'errata-restrict-installable': 0,
+                'content-view-id': self.content_view['id'],
+                'lifecycle-environment-id': self.env['id'],
+                'organization-id': self.org['id'],
+                'per-page': 1000,
+            }
+        )
+        for i in range(len(FAKE_9_YUM_ERRATUM)):
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_errata.py:531]: `range(len())` is considered anti-pattern in Python. Just go over values of iterable.
Also, `Erratum.list()` returns Python list, doesn't it? Please don't throw away more convenient data structure by converting it to string.

I'd do something like below, after confirming that `Erratum.list()` returns list of dicts, and that "Title" is the value we want to assert on. (Another candidate would be "Errata ID" - I don't have environment set up to check what they are.)
```python
erratum_title_list = [erratum['Title'] for erratum in erratum_list]
for erratum in FAKE_9_YUM_ERRATUM:
    assert erratum in erratum_title_list
```

Same comment applies to other usages of this pattern below.

File: tests/foreman/cli/test_errata.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:503]: Hi @mirekdlugosz , I will go with the first idea. Thank you.

File: tests/foreman/cli/test_errata.py (hunk 2 above)
Comment (swadeley) [reply tests/foreman/cli/test_errata.py:531]: will change those, using erratum_id_list
thank you

Comment (swadeley) [reply tests/foreman/cli/test_errata.py:531]: and key is errata-id

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/cli/test_errata.py -k test_positive_list_affected_chosts_by_erratum_restrict_flag
========================= test session starts ====================================
platform linux -- Python 3.7.7, pytest-4.6.3, py-1.8.1, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/virtwho/robottelo
plugins: cov-2.8.1, services-1.3.1, mock-1.10.4, forked-1.1.3, xdist-1.31.0
collecting ... 2020-05-26 11:49:57 - conftest - DEBUG - Collected 34 test cases
2020-05-26 12:49:58 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7f01f94bbdd0
2020-05-26 12:49:58 - robottelo.ssh - INFO - Connected to [dell-r330-12.gsslab.brq.redhat.com]
2020-05-26 12:49:58 - robottelo.ssh - INFO - >>> rpm -q satellite
2020-05-26 12:50:00 - robottelo.ssh - INFO - <<< stdout
satellite-6.8.0-0.3.beta.el7sat.noarch

2020-05-26 12:50:00 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f01f94bbdd0
2020-05-26 12:50:00 - robottelo.host_info - DEBUG - Host Satellite version: 6.8
2020-05-26 12:50:00 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 34 items / 33 deselected / 1 selected                                                                                            

tests/foreman/cli/test_errata.py .                                                                                                   [100%]

===================== 1 passed, 33 deselected in 830.75 seconds ===================

**Comment by jameerpathan111** [comment]
> ===================== 1 passed, 33 deselected in 830.75 seconds ===================

@swadeley we should avoid hostname and ip address in our public repos.

**Comment by swadeley** [comment]
Thank you

**Pull request merged**